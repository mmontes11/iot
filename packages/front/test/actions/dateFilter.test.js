import {
  toggleDateFilterType,
  selectTimePeriod,
  updateTimePeriod,
  updateStartDate,
  updateEndDate,
} from "actions/dateFilter";

describe("actions/dateFilter", () => {
  it("dispatches toggleDateFilterType action", () => {
    const thunk = toggleDateFilterType();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
  it("dispatches selectTimePeriod action", () => {
    const thunk = selectTimePeriod();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
  it("dispatches updateTimePeriod action", () => {
    const thunk = updateTimePeriod();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
  it("dispatches updateStartDate action", () => {
    const thunk = updateStartDate();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
  it("dispatches updateEndDate action", () => {
    const thunk = updateEndDate();
    const dispatch = jest.fn();
    thunk(dispatch);
    expect(thunk).toBeAThunk();
    expect(thunk).toMatchSnapshot();
  });
});
