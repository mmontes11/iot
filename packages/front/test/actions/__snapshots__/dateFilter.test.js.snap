// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/dateFilter dispatches selectTimePeriod action 1`] = `[Function]`;

exports[`actions/dateFilter dispatches toggleDateFilterType action 1`] = `[Function]`;

exports[`actions/dateFilter dispatches updateEndDate action 1`] = `[Function]`;

exports[`actions/dateFilter dispatches updateStartDate action 1`] = `[Function]`;

exports[`actions/dateFilter dispatches updateTimePeriod action 1`] = `[Function]`;
